pkg/k8s/endpointsynchronizer/cep.go

```diff
@@ -225,48 +227,72 @@ func (epSync *EndpointSynchronizer) RunK8sCiliumEndpointSync(e *endpoint.Endpoin
 					return nil
 				}
 
-				// We have no localCEP copy. We need to fetch it for updates, below.
-				// This is unexpected as there should be only 1 writer per CEP, this
-				// controller, and the localCEP created on startup will be used.
-				if localCEP == nil {
-					localCEP, err = ciliumClient.CiliumEndpoints(namespace).Get(podName, meta_v1.GetOptions{})
-					switch {
-					// The CEP doesn't exist in k8s. This is unexpetected but may occur
-					// if the endpoint was removed from k8s but not yet within the agent.
-					// Mark the CEP for creation on the next controller iteration. This
-					// may never occur if the controller is stopped on Endpoint delete.
-					case err != nil && k8serrors.IsNotFound(err):
-						needInit = true
+				switch {
+				case k8s.JSONPatchVerConstr.Check(k8sServerVer):
```

Comment:
:( I was hoping this case and the code with it would go in the same switch where we decide on Update or UpdateStatus. Is the reason it's here because it doesn't need localCEP? (and so there is no need to do the other handling below to re-fetch it on errors etc.)
If so, it might be harmless to keep localCEP updated but not actually use it (the API for Patch and Update already sent it back and we parsed it, right? we just assign it here or not)